### ISSUE TITLE
VAE onset loss is non-differentiable and double-counts the KL term

### ISSUE THREAD

**Issue by p3zo** [p3zo/rhythmic-relationships#16]
```python
# scripts/modeling/part_vae/train_part_vae.py:77-82
x_recon_binary = (x_recon > 0).to(torch.float32)
onset_loss = compute_recon_loss(x_recon_binary, x_binary, mu, sigma, loss_fn)
velocity_loss = compute_recon_loss(x_recon, x, mu, sigma, loss_fn)
loss = onset_loss + velocity_loss
```

`(x_recon > 0)` is a non-differentiable threshold, so the tensor handed to the loss has no grad history. The reconstruction half of `onset_loss` is therefore a **constant** — it contributes nothing to the gradient.

What does survive is the KL term, because `compute_recon_loss` adds `kld_loss` internally. So the net effect of adding `onset_loss` is:

- onset reconstruction: trains nothing
- KL divergence: weighted **2x**

Same code in `scripts/modeling/part_pair_vae/train_part_pair_vae.py:68-73`.

**Fix**: separate the KL term from the reconstruction term so it's added exactly once, and compute the onset loss on the raw logits against the binarised target (`loss_fn(x_recon, x_binary)`) rather than on a thresholded prediction.
